packages/http-client-java/generator/http-client-generator-test/src/main/java/tsptest/multipleapiversion/models/Resource.java

```diff
@@ -21,7 +21,7 @@ public final class Resource implements JsonSerializable<Resource> {
      * The id property.
      */
     @Generated
-    private String id;
+    private final String id;
```

Comment:
There seems a minor change on "required", not sure where it comes from.

The typespec in local test is "required"
```
  @resource("resources")
  model Resource {
    @visibility(Lifecycle.Read)
    id: string;

    @key
    @visibility(Lifecycle.Read)
    name: string;

    type: string;
  }
```

Reply:
Em, the problem seems on the "read-only".

If ctor is public, user would have to put an `id` when this is read-only.

Reply:
OK, the problem seems only happen when there were more than 1 `@service`.

Ignore this, as such case should not happen to Azure yet.